crates/core/src/entry/validation_dependencies.rs

```diff
@@ -0,0 +1,115 @@
+use crate::network::entry_with_header::EntryWithHeader;
+use holochain_core_types::entry::Entry;
+use holochain_persistence_api::cas::content::Address;
+
+pub trait ValidationDependencies {
+    fn get_validation_dependencies(&self) -> Vec<Address>;
+}
+
+impl ValidationDependencies for EntryWithHeader {
+    fn get_validation_dependencies(&self) -> Vec<Address> {
+        match &self.entry {
+            Entry::App(_, _) => {
+                // in the future an entry should be dependent on its header but 
+                // for now it can require nothing
+                Vec::new()
```

Comment:
If the entry is an update, which we can tell from the header (`self.header.crud_link.is_some()`) that should be the dependency.

Reply:
Good catch, thanks. It is kind of nice at the moment that these are only used for improving the ordering. If we don't include the correct dependencies it just falls back to how it worked originally